be/src/exec/vectorized/json_scanner.cpp

```diff
@@ -68,20 +68,32 @@ Status JsonScanner::open() {
 StatusOr<ChunkPtr> JsonScanner::get_next() {
     SCOPED_RAW_TIMER(&_counter->total_ns);
     ChunkPtr src_chunk;
-    RETURN_IF_ERROR(_create_src_chunk(&src_chunk));
 
-    if (_cur_file_eof) {
-        RETURN_IF_ERROR(_open_next_reader());
-        _cur_file_eof = false;
-    }
-    Status status = _cur_file_reader->read_chunk(src_chunk.get(), _max_chunk_size, _src_slot_descriptors);
-    if (status.is_end_of_file()) {
-        _cur_file_eof = true;
-    }
+    // Read until we get a non-empty chunk.
+    do {
+        RETURN_IF_ERROR(_create_src_chunk(&src_chunk));
```

Comment:
This can be put out of the while block.

Reply:
I think ` _cur_file_reader->read_chunk` do not guarantee the src_chunk is not modified when the `src_chunk->num_rows == 0`. Besides, trom the performance perspective, more than one chunk would be created only when the `get_next()` returns error.